WordPress/gutenberg
# Move default template types and template part areas to REST API

## What?


This PR moves the `defaultTemplateTypes`, `defaultTemplatePartAreas`  to the `getEntityRecord( 'root', '__unstableBase' ) endpoint`. 



 Given that the impacted selectors aren't specific to the `editor` store, I decided to deprecate them all. I couldn't find a suitable location to ensure accessibility across different packages, leading to some duplicated code. Should we create a `@wordpress/template` package (similar to `@wordpress/patterns`) to house this logic? Create a dedicated package could help us with #65390 too.

## Why

This PR is necessary to remove the `core/editor` dependency across the application. In the specific, this refactor is needed to allow the package `fields` to have access to the `defaultTemplateTypes`, `defaultTemplatePartAreas` without that editor settings are initialized. More details here: https://github.com/WordPress/gutenberg/pull/65390#discussion_r1814724405

## Testing Instructions





1. Visit the Site Editor. 
2. On the sidebar click on `Pattern`.
3. Click on `All template parts`.
4. Ensure that all the patterns are visible (you should see `Header` and `Footer` labels too)
5. On the top right, click on `Add New Pattern`.
6. Create a new template part
7. Ensure that the template part is created correctly.
8. Visit the Site Editor.
9. Visit the `Blog Home`.
10. Ensure that the right panel under the template section renders the right data:
![image](https://github.com/user-attachments/assets/67316a2d-6b21-42ca-b1f1-7c7795499367)

11. Add a block into the footer.
12. Click save.
13. Ensure that `are you ready to save` shows the right data:

![image](https://github.com/user-attachments/assets/e3b29415-c761-4d1e-aa4c-c72307ab59a0)

### Testing Instructions for Keyboard


## Screenshots or screencast
